### PR TITLE
Menu improvements, scene fading, and pre-level message generator

### PR DESCRIPTION
**Pull request by AleksandrHovhannisyan**
Open Menu.unity and run to test.

Major changes:

- Scene switching relies on scene names as opposed to indices. This makes it much clearer to follow the flow of the game, since indices may change and are not all that informative.

- Menu button font changed. Menu button normal color changed so the contrast between highlighted and un-highlighted is greater.

- Added a scene fader prefab for smoother scene transitions on the menu.

- Added an intro scene that, by default, loads Level1 (other levels can then dictate what the next level transition will be). This scene is to be called between each level transition. The scene plays a dramatic audio cue and displays a random message.

### CODE REVIEW

File: Assets/Scripts/Menus/MenuMusic.cs [hunk 1]
```diff
@@ -9,10 +9,7 @@ public class MenuMusic : MonoBehaviour
     // Singleton design pattern to avoid audio duplication
     private static MenuMusic instance = null;
     public static MenuMusic Instance { get { return instance; } }
-
-    // Override via the editor as needed if build settings change
-    public int menuIndex = 0;
-    public int controlsIndex = 1;
+    public static AudioSource GetMusic() { return Instance.GetComponent<AudioSource>(); }
```
Comment (klein1992) [inline Assets/Scripts/Menus/MenuMusic.cs:12]: lol I wasn't aware of this method. I just added an audio source for my level change scene.

Comment (AleksandrHovhannisyan) [reply Assets/Scripts/Menus/MenuMusic.cs:12]: Not sure what you mean, that's what I did as well :)

Comment (klein1992) [reply Assets/Scripts/Menus/MenuMusic.cs:12]: Basically, I didnt use this method to place the music on my scene (duplicating audio source).

File: Assets/Scripts/Menus/LoadSceneOnClick.cs [hunk 2]
```diff
@@ -1,22 +1,50 @@
...
+        else
+        {
+            menuMusic = MenuMusic.GetMusic();
+            StartCoroutine(LoadSceneDelayed(sceneName));
+        }
+    }
+
+
+    /* Delays scene transition to allow for menu music to fade.
+     */
```
Comment (klein1992) [inline Assets/Scripts/Menus/LoadSceneOnClick.cs:35]: Not sure how I am supposed to use this method. Let's say I want to load Scene "Level2" once the player has activated my trap door. How would I take advantage of this method?

**Review by klein1992 — APPROVED** [review]
Looks good to me just leaving some comments

**Comment by klein1992** [comment]
Please note that the camera view needs to be changed. I can't see the full Embody Menu screen now due to your changes.

File: Assets/Scripts/Menus/MenuMusic.cs (hunk 1 above)
Comment (AleksandrHovhannisyan) [reply Assets/Scripts/Menus/MenuMusic.cs:12]: I should clarify :D There's no duplication; this is just returning a reference to the audio component.

File: Assets/Scripts/Menus/LoadSceneOnClick.cs (hunk 2 above)
Comment (AleksandrHovhannisyan) [reply Assets/Scripts/Menus/LoadSceneOnClick.cs:35]: Well, this is technically not for that, this is for menu buttons.

**Pull request merged**